python-avd/pyavd/_cv/client/versioning.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from .constants import CVAAS_VERSION_STRING
+
+
+class CvVersion:
+    version: str
+    version_value: int
+
+    def __init__(self, version: str) -> None:
+        self.version = version
+        self.version_value = self._version_as_value()
+
+    def __str__(self) -> str:
+        return self.version
+
+    def __repr__(self) -> str:
+        return f"<CvVersion {self.version}>"
+
+    def __hash__(self) -> int:
+        return self.version_value
+
+    def _version_as_value(self) -> int:
+        if self.version == CVAAS_VERSION_STRING:
+            return 99999999
+
+        elements = self.version.split(".", maxsplit=2)
+        return int(f"{elements[0]:04}{elements[1]:02}{elements[2]:02}")
```

Comment:
this can be quite fragile though I guess we are not expecting anyone but pyavd to use the method so thats probably ok

Reply:
fixed.